docs/user/filter-and-process/custom-pipeline-processing-ottl.md

```diff
@@ -0,0 +1,161 @@
+2. **Filters (filter)**: The transformed data is then evaluated against the rules in the `filter` section. Any data matching a filter condition is dropped
+
+This sequence ensures that you filter on the final, transformed state of your data. For example, if you rename an attribute in a transform rule, your filter rule must use the new name, not the original one.
+
+## OTTL Context and Data Processing
+
+Transform and filter operations work at different context levels depending on the telemetry data type. Currently, custom context configuration is not supported, and all operations use the lowest-level context by default:
+
+| Pipeline Type  | Default Context | Description                                       | Requirements              |
+| :------------- | :-------------- | :------------------------------------------------ | :------------------------ |
+| LogPipeline    | `log`           | Operations apply to individual log records        | OTLP output only          |
+| MetricPipeline | `datapoint`     | Operations apply to individual metric data points | Available for all outputs |
+| TracePipeline  | `span`          | Operations apply to individual trace spans        | Available for all outputs |
+
+> **Note**: For TracePipeline, the `span` context is used instead of `spanevent` because traces can contain spans without span events. This means you cannot currently filter or transform data within the `spanevent` context for traces.
+
+When writing OTTL expressions, you must include the appropriate context path to access data at different levels:
+
+- **Access current context**: `log.attributes["key"]`, `datapoint.value`, `span.name`
+- **Access higher contexts**: `resource.attributes["key"]`, `scope.name`, `metric.name` (from datapoint context)
+- **Access related contexts**: `span.attributes["key"]` (from spanevent context, when available)
+
+> **Note**: By default, if an OTTL statement encounters an error, the processor logs the error and continues to process the next piece of data. This `ignore` mode prevents a single malformed data point from halting the entire pipeline.
+
+## Transforming Telemetry Data
+
+Use transformations to modify telemetry data as it flows through a pipeline. You can add, update, or delete attributes, change metric types, or modify span details to enrich data or conform to a specific schema.
+
+You configure transformations in the `transform` section of a pipeline's spec.
+
+Each transformation rule consists of:
+
+- **statements**: A list of OTTL functions to execute
+- **conditions** (optional): A list of OTTL conditions. If you provide conditions, the statements only run if at least one of the conditions is true
+
+### Example: General Resource Attribute Enrichment
+
+This example adds a `deployment.environment` attribute with the value `production` to all metrics in the pipeline. Since there are no conditions, the rule applies to all data.
```

Comment:
```suggestion
This example adds a `deployment.environment.name` attribute with the value `production` to all metrics in the pipeline. Since there are no conditions, the rule applies to all data.
```
Just to have an example conform to https://opentelemetry.io/docs/specs/semconv/resource/deployment-environment/